2020-12-18 11:51:47,659 ERROR: Exception on /index [POST] [in /home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/routes.py", line 34, in index
    post = Post(body=form.post.data, author=current_user)
NameError: name 'Post' is not defined
2020-12-18 11:53:26,690 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:48]
2020-12-18 12:18:16,511 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:48]
2020-12-18 12:18:24,127 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:48]
2020-12-18 12:43:52,446 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:48]
2020-12-18 13:16:41,434 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 13:23:02,747 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 13:41:10,249 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 13:55:03,622 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 14:19:43,917 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 15:48:49,556 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 15:49:07,638 ERROR: Exception on /reset_password_request [GET] [in /home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/routes.py", line 176, in reset_password_request
    form = ResetPasswordRequestForm()
NameError: name 'ResetPasswordRequestForm' is not defined
2020-12-18 15:51:17,085 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 15:51:21,554 ERROR: Exception on /reset_password_request [GET] [in /home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/routes.py", line 177, in reset_password_request
    form = ResetPasswordRequestForm()
NameError: name 'ResetPasswordRequestForm' is not defined
2020-12-18 15:53:50,688 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 15:55:23,185 ERROR: Exception on /reset_password_request [POST] [in /home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mahmoudodeh/Desktop/Flask/flask-venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/routes.py", line 181, in reset_password_request
    send_password_reset_email(user)
NameError: name 'send_password_reset_email' is not defined
2020-12-18 15:57:00,470 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:14:17,900 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:20:05,969 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:21:55,863 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:26:55,474 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:28:01,243 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:34:19,550 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 16:41:55,857 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 17:25:59,613 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 17:32:22,571 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 20:35:08,876 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
2020-12-18 20:36:14,288 INFO: Microblog startup [in /home/mahmoudodeh/Desktop/Flask/Flask-Tutorial/microblog/app/__init__.py:50]
